runtime/domain.c

```diff
@@ -355,11 +355,11 @@ void caml_init_domains(uintnat minor_heap_wsz) {
     caml_fatal_error("Minor_heap_max misconfigured for this platform");
 
   /* reserve memory space for minor heaps and tls_areas */
-  size = (uintnat)Minor_heap_max * Max_domains;
+  size = (uintnat)Bsize_wsize(Minor_heap_max) * Max_domains;
```

Comment:
It would be better to rename `Minor_heap_max` to `Minor_heap_max_wsize`. And add a comment where `Minor_heap_max_wsize` is defined to say that this would allow for a maximum minor heap size of 2GB.